ui-admin/src/portal/siteContent/designer/editors/ButtonsEditor.tsx

```diff
@@ -0,0 +1,131 @@
+import { ButtonConfig, HtmlSection, SectionConfig } from '@juniper/ui-core'
+import React, { useId } from 'react'
+import classNames from 'classnames'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faChevronDown, faChevronUp, faPlus } from '@fortawesome/free-solid-svg-icons'
+import Select from 'react-select'
+import { TextInput } from 'components/forms/TextInput'
+import { Button } from 'components/forms/Button'
+
+/**
+ * Returns an editor allowing the user to add and edit buttons
+ */
+export const ButtonsEditor = ({ section, updateSection }: {
+    section: HtmlSection, updateSection: (section: HtmlSection) => void
+}) => {
+  const config = JSON.parse(section.sectionConfig || '{}') as SectionConfig
+  const buttons = config.buttons as ButtonConfig[] || []
+  const buttonsContentId = useId()
+  const buttonsTargetSelector = `#${buttonsContentId}`
+  return (
+    <div>
+      <div className="pb-1">
+        <button
+          aria-controls={buttonsTargetSelector}
+          aria-expanded="false"
+          className={classNames('btn w-100 py-2 px-0 d-flex text-decoration-none')}
+          data-bs-target={buttonsTargetSelector}
+          data-bs-toggle="collapse"
+        >
+          <span className={'form-label fw-semibold mb-0'}>Buttons ({buttons.length})</span>
+          <span className="text-center px-2">
+            <FontAwesomeIcon icon={faChevronDown} className="hidden-when-collapsed"/>
+            <FontAwesomeIcon icon={faChevronUp} className="hidden-when-expanded"/>
+          </span>
+        </button>
+      </div>
+      <div className="collapse hide rounded-3 mb-2" id={buttonsContentId}
+        style={{ backgroundColor: '#eee', padding: '0.75rem' }}>
+        <div>
+          {buttons.map((button, i) => {
+            return <ButtonEditor key={i} button={button} updateButton={newButton => {
+              const newButtons = [...buttons]
+              newButtons[i] = newButton
+              updateSection({ ...section, sectionConfig: JSON.stringify({ ...config, buttons: newButtons }) })
+            }}/>
+          })}
+        </div>
+        <Button onClick={() => {
+          const newButtons = [...buttons]
+          newButtons.push({ type: 'internalLink', text: '', href: '' })
+          updateSection({ ...section, sectionConfig: JSON.stringify({ ...config, buttons: newButtons }) })
+        }}><FontAwesomeIcon icon={faPlus}/> Add Button
+        </Button>
+      </div>
+    </div>
+  )
+}
+
+/**
+ * Returns an editor for a single button
+ */
+export const ButtonEditor = ({ button, updateButton }: {
```

Comment:
Is there some way to adjust the styling? It seems weird that the styling is different based on the button type but you have no control over it. It seems like that was already baked in, though, so maybe the subject of a later PR.

Reply:
I'll add a ticket to the backlog- I totally agree that it would be great to be able to choose which button style to use